bin/api-docs/update-api-docs.js

```diff
@@ -0,0 +1,233 @@
+/**
+ * External dependencies
+ */
+const { join, relative, resolve, sep, basename, dirname } = require( 'path' );
+const glob = require( 'fast-glob' );
+const execa = require( 'execa' );
+const { Transform } = require( 'stream' );
+const { readFile } = require( 'fs' ).promises;
+
+/**
+ * README file tokens, defined as a tuple of token identifier, source path.
+ *
+ * @typedef {[string,string]} WPReadmeFileTokens
+ */
+
+/**
+ * README file data, defined as a tuple of README file path, token details.
+ *
+ * @typedef {[string,WPReadmeFileTokens]} WPReadmeFileData
+ */
+
+/**
+ * Path to root project directory.
+ *
+ * @type {string}
+ */
+const ROOT_DIR = resolve( __dirname, '../..' );
+
+/**
+ * Path to packages directory.
+ *
+ * @type {string}
+ */
+const PACKAGES_DIR = resolve( ROOT_DIR, 'packages' );
+
+/**
+ * Path to data documentation directory.
+ *
+ * @type {string}
+ */
+const DATA_DOCS_DIR = resolve(
+	ROOT_DIR,
+	'docs/designers-developers/developers/data'
+);
+
+/**
+ * Pattern matching start token of a README file.
+ *
+ * @type {RegExp}
+ */
+const TOKEN_PATTERN = /<!-- START TOKEN\((.+?(?:\|(.+?))?)\) -->/g;
+
+/**
+ * Given an absolute file path, returns the package name.
+ *
+ * @param {string} file Absolute path.
+ *
+ * @return {string} Package name.
+ */
+function getFilePackage( file ) {
+	return file.startsWith( PACKAGES_DIR )
+		? relative( PACKAGES_DIR, file ).split( sep )[ 0 ]
+		: getDataDocumentationFilePackage( basename( file, '.md' ) );
+}
+
+/**
+ * Returns an appropriate glob pattern for the packages directory to match
+ * relevant documentation files for a given set of files.
+ *
+ * @param {string[]} files Set of files to match. Pass an empty set to match
+ *                         all packages.
+ *
+ * @return {string} Packages glob pattern.
+ */
+function getPackagePattern( files ) {
+	if ( ! files.length ) {
+		return '*';
+	}
+
+	// Since brace expansion doesn't work with a single package, special-case
+	// the pattern for the singular match.
+	const packages = Array.from( new Set( files.map( getFilePackage ) ) );
+	return packages.length === 1 ? packages[ 0 ] : '{' + packages.join() + '}';
+}
+
+/**
+ * Returns the conventional store name of a given package.
+ *
+ * @param {string} packageName Package name.
+ *
+ * @return {string} Store name.
+ */
+function getPackageStoreName( packageName ) {
+	let storeName = 'core';
+	if ( packageName !== 'core-data' ) {
+		storeName += '/' + packageName;
+	}
+
+	return storeName;
+}
+
+/**
+ * Returns the conventional package name of a given documentation file name.
+ *
+ * @param {string} file Documentation file name.
+ *
+ * @return {string} package name.
+ */
+function getDataDocumentationFilePackage( file ) {
```

Comment:
Following up on https://github.com/WordPress/gutenberg/pull/18840/files#r396461733

In which use cases this function is called?

Reply:
Likely the same response from https://github.com/WordPress/gutenberg/pull/18840#discussion_r396771738 😅 I had a feeling that there was some further simplification I might have been able to do after 37adca35027a1ba3bf7d16ecaff6b65c6772ba94, but I clearly didn't follow the logic thoroughly enough to see where those simplifications could be made.

Reply:
https://github.com/WordPress/gutenberg/pull/18840/commits/77dd061a875031d22bd87901c1461333e0a1c49e